saleor/discount/models.py

```diff
@@ -33,7 +33,9 @@ def __init__(self, msg, min_amount_spent=None, min_checkout_items_quantity=None)
 class VoucherQueryset(models.QuerySet):
     def active(self, date):
         return self.filter(
-            Q(usage_limit__isnull=True) | Q(used__lt=F("usage_limit")),
+            Q(usage_limit__isnull=True)
+            | Q(used__lt=F("usage_limit"))
+            | Q(usage_limit=0),
```

Comment:
We should add some notes in docs or in the dashboard. I assume that some user could change usage_limit to 0 to cancel voucher. They could think that setting this to 0 will block voucher usage.

Reply:
Maybe a better approach is to get Null from Fronted? It could be more intuitive.